examples/dht_server.rs

```diff
@@ -155,21 +157,27 @@ fn main() {
     // The server task asynchronously iterates over and processes each
     // incoming packet.
     let server_obj_c = server_obj.clone();
-    let network_reader = stream.for_each(move |(packet, addr)| {
+    let network_reader = stream.then(Ok).filter(|packet| // TODO: use filter_map from futures 2 to avoid next `expect`
+        match packet {
+            Ok(_) => true,
+            Err(e) => {
+                error!("packet receive error = {:?}", e);
+                // ignore packet decode errors
+                e.cause().downcast_ref::<DecodeError>().is_none()
+            }
+        }
+    ).then(|packet: Result<_, ()>|
+        packet.expect("always ok")
+    ).for_each(move |(packet, addr)| {
         println!("recv = {:?}", packet.clone());
         server_obj_c.handle_packet(packet, addr).or_else(|err| {
             error!("failed to handle packet: {:?}", err);
             future::ok(())
         })
-    }).map_err(|err| {
-        // All tasks must have an `Error` type of `()`. This forces error
-        // handling and helps avoid silencing failures.
-        error!("packet receive error = {:?}", err);
-        Error::new(ErrorKind::Other, "udp receive error")
-    });
+    }).map_err(|e| Error::new(ErrorKind::Other, e.compat()));
 
     let network_writer = rx
-        .map_err(|_| Error::new(ErrorKind::Other, "rx error"))
+        .map_err(|_| err_msg("rx error"))
```

Comment:
let it be done in another huge PR when we will move to crate failure to use nice errs instead of string errors